cmd/build/v2/build.go

```diff
@@ -155,15 +163,19 @@ func (bc *OktetoBuilder) Build(ctx context.Context, options *types.BuildOptions)
 				oktetoLog.SetStage(fmt.Sprintf("Building service %s", svcToBuild))
 			}
 
-			// TODO: uncomment this when we have the commit + build args hashed
-			// if imageTag, isBuilt := bc.checkIfCommitIsAlreadyBuilt(options.Manifest.Name, svcToBuild, bc.Config.GetHash(), options.NoCache); isBuilt {
-			// 	oktetoLog.Warning("Skipping build of '%s' image because it's already built for commit %s", svcToBuild, bc.Config.GetHash())
-			// 	bc.SetServiceEnvVars(svcToBuild, imageTag)
-			// 	bc.builtImages[svcToBuild] = true
-			// 	continue
-			// }
-
 			buildSvcInfo := buildManifest[svcToBuild]
+
+			// We only check that the image is built in the global registry if the noCache option is not set
+			if !options.NoCache && bc.Config.IsCleanProject() {
```

Comment:
I think `bc.Config.IsCleanProject()` doesn't rely on the image so I think we caoul extract this func before the `for` loop?

Reply:
The `IsCleanProject()` was precalculated and is just a boolean. We can move it before but it's already calculated when the user starts a build